index.js

```diff
@@ -1,20 +1,606 @@
-const readline = require('readline');
-const readlineInterface = readline.createInterface(process.stdin, process.stdout);
+//Boilerplate code set up correctly - used to accept input from user
+const { ENETRESET } = require("constants");
+const { read } = require("fs");
+const { exit } = require("process");
+const readline = require("readline");
+const readlineInterface = readline.createInterface(
+  process.stdin,
+  process.stdout
+);
 
 function ask(questionText) {
   return new Promise((resolve, reject) => {
     readlineInterface.question(questionText, resolve);
   });
 }
 
-start();
+//------------------------------Player------------------------------//
+//defines an object to hold information about the player
+let player = {
+  name: "Player1",
+  inventory: [],
+  currentRoom: "",
+};
+
+//------------------------------Items------------------------------//
+//defines a class of items - constructors include
+// - name (full name of the item)
+// - descriptionTxt (text about the object when the player interacts with it)
+// - readTxt (text about the object when the player reads it)
+// - greetTxt (text about the object when the player greets it)
+// - takeable (boolean indicating if the object is takable or not)
+class checkInventory {
+  constructor(name, descriptionTxt, readTxt, greetTxt, takeable) {
+    this.name = name;
+    this.descriptionTxt = descriptionTxt;
+    this.readTxt = readTxt;
+    this.greetTxt = greetTxt;
+    this.takeable = takeable;
+  }
+  examine() {
+    return this.descriptionTxt;
+  }
+  read() {
+    return this.readTxt;
+  }
+
+  greet() {
+    return this.greetTxt;
+  }
+
+  take() {
+    if (this.takeable) {
+      //adds item to players inventory array
+      player.inventory.push(this.name);
```

Comment:
And once you check to see that the item exists in the room you are in, you can do the opposite of line 54 and use your `.pop()` method to remove the item from the room inventory and put that item in the player inventory. This would stop you from being able to take the items multiple times each